custom/aaa/models.py

```diff
@@ -453,16 +489,80 @@ def agg_from_household_case_ucr(cls, domain, window_start, window_end):
             household_cases_ucr_tablename=ucr_tablename,
         ), {'domain': domain, 'window_start': window_start, 'window_end': window_end}
 
+    @classmethod
+    def agg_from_birth_preparedness_forms_ucr(cls, domain, window_start, window_end):
+        doc_id = StaticDataSourceConfiguration.get_doc_id(domain, 'reach-birth_preparedness')
+        config, _ = get_datasource_config(doc_id, domain)
+        ucr_tablename = get_table_name(domain, config.table_id)
+
+        return """
+        UPDATE "{ccs_record_tablename}" AS ccs_record SET
+            num_anc_checkups = bp_forms.num_anc_checkups
+        FROM (
+            SELECT ccs_record_case_id, COUNT(*) as num_anc_checkups
+            FROM "{bp_forms_ucr_tablename}"
+            WHERE ccs_record_case_id IS NOT NULL
+            GROUP BY ccs_record_case_id
+        ) bp_forms
+        WHERE ccs_record.ccs_record_case_id = bp_forms.ccs_record_case_id
+        """.format(
+            ccs_record_tablename=cls._meta.db_table,
+            bp_forms_ucr_tablename=ucr_tablename,
+        ), {'domain': domain, 'window_start': window_start, 'window_end': window_end}
+
     @classproperty
     def aggregation_queries(self):
         return [
             self.agg_from_ccs_record_case_ucr,
             self.agg_from_person_case_ucr,
             self.agg_from_household_case_ucr,
+            self.agg_from_birth_preparedness_forms_ucr,
             self.agg_from_village_ucr,
             self.agg_from_awc_ucr,
         ]
 
+    def birth_preparedness_form_details(self):
+        doc_id = StaticDataSourceConfiguration.get_doc_id(self.domain, 'reach-birth_preparedness')
+        config, _ = get_datasource_config(doc_id, self.domain)
+        ucr_tablename = get_table_name(self.domain, config.table_id)
+
+        with connections['aaa-data'].cursor() as cursor:
+            cursor.exectue(
+                'SELECT * FROM "{}" WHERE ccs_record_case_id = %s'.format(ucr_tablename),
+                [self.ccs_record_case_id]
+            )
+            result = dictfetchall(cursor)
+
+        return result
+
+    def postnatal_care_form_details(self):
+        doc_id = StaticDataSourceConfiguration.get_doc_id(self.domain, 'reach-postnatal_care')
+        config, _ = get_datasource_config(doc_id, self.domain)
+        ucr_tablename = get_table_name(self.domain, config.table_id)
+
+        with connections['aaa-data'].cursor() as cursor:
+            cursor.exectue(
+                'SELECT * FROM "{}" WHERE ccs_record_case_id = %s'.format(ucr_tablename),
+                [self.ccs_record_case_id]
+            )
+            result = dictfetchall(cursor)
+
+        return result
```

Comment:
Will be trying a different method for data retrieval with these where I pull directly from the UCR for data that does not need to be aggregated or displayed in filters